mlrun/api/api/endpoints/workflows.py

```diff
@@ -0,0 +1,320 @@
+    project: str,
+    name: str,
+    request: fastapi.Request,
+    workflow_request: mlrun.common.schemas.WorkflowRequest = mlrun.common.schemas.WorkflowRequest(),
+    auth_info: mlrun.common.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: Session = fastapi.Depends(mlrun.api.api.deps.get_db_session),
+):
+    """
+    Submitting a workflow of existing project.
+    To support workflow scheduling, we use here an auxiliary function called 'load_and_run'.
+    This function runs remotely (in a distinct pod), loads a project and then runs the workflow.
+    In this way we can run the workflow remotely with the workflow's engine or
+    schedule this function which in every time loads the project and runs the workflow.
+    Notice:
+    in case of simply running a workflow, the returned run_id value is the id of the run of the auxiliary function.
+    For getting the id and status of the workflow, use the `get_workflow_id` endpoint with the returned run id.
+
+    :param project:             name of the project
+    :param name:                name of the workflow
+    :param request:             fastapi request for supporting rerouting to chief if needed
+    :param workflow_request:    the request includes: workflow spec, arguments for the workflow, artifact path
+                                as the artifact target path of the workflow, source url of the project for overriding
+                                the existing one, run name to override the default: 'workflow-runner-<workflow name>'
+                                and kubernetes namespace if other than default
+    :param auth_info:           auth info of the request
+    :param db_session:          session that manages the current dialog with the database
+
+    :returns: response that contains the project name, workflow name, name of the workflow,
+             status, run id (in case of a single run) and schedule (in case of scheduling)
+    """
+    project = await run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().get_project,
+        db_session=db_session,
+        name=project,
+        leader_session=auth_info.session,
+    )
+
+    # check permission CREATE run
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        resource_type=mlrun.common.schemas.AuthorizationResourceTypes.run,
+        project_name=project.metadata.name,
+        resource_name=workflow_request.run_name or "",
+        action=mlrun.common.schemas.AuthorizationAction.create,
+        auth_info=auth_info,
+    )
+    # check permission READ workflow
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        resource_type=mlrun.common.schemas.AuthorizationResourceTypes.workflow,
+        project_name=project.metadata.name,
+        resource_name=name,
+        action=mlrun.common.schemas.AuthorizationAction.read,
```

Comment:
Why read workflow? Aren't we creating a new workflow?

Reply:
Not exactly.. The workflows are not stored in the database, and only exists in the project itself (Maybe need to refactor all this behavior in the future), hence only read permission here.

Reply:
I understand what you're saying in an implementation point of view.
But if I POST to submit_workflow, I am creating a workflow, the resource would be in kfp and not in mlrun. But it's still a create operation. Unless you are saying that the workflow definitely already existed before POSTing to submit_workflow?